pkg/cmd/pulumi/convert-trace.go

```diff
@@ -648,6 +648,7 @@ func (t *otelTrace) newSpan(root *appdash.Trace, parent *otelSpan) error {
 
 func (t *otelTrace) getNextSpanID() trace.SpanID {
 	var id trace.SpanID
+	//nolint:gosec // len is always positive
 	binary.BigEndian.PutUint64(id[:], uint64(len(t.spans)+1))
```

Comment:
if you rewrite this as 
```suggestion
	binary.BigEndian.PutUint64(id[:], uint64(len(t.spans))+1)
```
Is the linter smart enough to see the len->uint64 cast and ignore it?

Reply:
That worked, nice!